static/app/views/explore/spans/quotaAlert.tsx

```diff
@@ -0,0 +1,146 @@
+  }
+
+  return null;
+}
+
+function useQuotaExceededAlertMessage() {
+  const location = useLocation();
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+  const {start, end, statsPeriod} = normalizeDateTimeParams(location.query);
+
+  const {data: performanceUsageStats} = usePerformanceUsageStats({
+    organization,
+    dateRange: statsPeriod ? {statsPeriod} : {start, end},
+    projectIds: selection.projects,
+  });
+
+  // Check if events were dropped due to exceeding the transaction/spans quota
+  const droppedEventsCount = performanceUsageStats?.totals['sum(quantity)'] || 0;
+
+  const {
+    data: {hasExceededPerformanceUsageLimit, subscription},
+  } = usePerformanceSubscriptionDetails();
+
+  if (droppedEventsCount === 0 || !hasExceededPerformanceUsageLimit || !subscription) {
+    return null;
+  }
+
+  const budgetType = subscription?.onDemandBudgets?.enabled
+    ? ['am1', 'am2'].includes(subscription.planTier)
+      ? t('on-demand budget')
+      : t('pay-as-you-go budget')
+    : t('reserved volumes');
+  const formattedDateRange: string | null = getFormattedDateTime({
+    statsPeriod,
+    start,
+    end,
+  });
+  const billingPageLink = (
+    <Link
+      to={{
+        pathname: `/settings/billing/checkout/?referrer=trace-view`,
+        query: {
+          skipBundles: true,
+        },
+      }}
+    >
+      {tct('increase your [budgetType]', {budgetType})}
+    </Link>
+  );
+
+  const subscriptionRenewalDate = dateFormatter(new Date(subscription.renewalDate));
+
+  if (!formattedDateRange) {
+    return tct(
+      'You’ve exceeded your [budgetType] during this date range and results will be skewed. We can’t collect more spans until [subscriptionRenewalDate]. If you need more, [billingPageLink]',
+      {
+        subscriptionRenewalDate,
+        billingPageLink,
+        budgetType,
+      }
+    );
+  }
+
+  return tct(
+    'You’ve exceeded your [budgetType] during this date range and results will be skewed. We can’t collect more spans until [subscriptionRenewalDate]. [rest]',
+    {
+      budgetType,
+      subscriptionRenewalDate,
+      rest: statsPeriod
+        ? tct(
+            'If you need more, [billingPageLink] or adjust your date range prior to [formattedDateRange].',
+            {
+              billingPageLink,
+              formattedDateRange,
+            }
+          )
+        : tct(
+            'If you need more, [billingPageLink] or adjust your date range before or after [formattedDateRange].',
+            {
+              billingPageLink,
+              formattedDateRange,
+            }
+          ),
+    }
+  );
+}
+
+export function QuotaExceededAlert() {
+  const message = useQuotaExceededAlertMessage();
+
+  if (!message) {
+    return null;
+  }
+
+  return (
+    <StyledAlert type="warning" showIcon>
+      {message}
+    </StyledAlert>
+  );
+}
+
+const StyledAlert = styled(Alert)`
+  margin-bottom: ${space(2)};
+`;
```

Comment:
I think we have `<Alert.Container>` for this functionality now.

Reply:
💯 @Abdkhan14 this should just be this layout going forward
```jsx
<Alert.Container>
   <Alert/>
</Alert.Container>
```

You can take a look at the alert docs [here](https://sentry.sentry.io/stories/?name=app%2Fcomponents%2Fcore%2Falert.stories.tsx&query=alert)

Reply:
Yup, thank you for the heads up folks!